File: java/algorithms/easy/simpleArraySum.java
/**
 * @author Rithie Natan Carvalhaes Prado
 * @date 2026-04-24
 * Last modified: 2026-06-11
 * @description Simple Array Sum - HackerRank
 * @see {@link https://www.hackerrank.com/challenges/simple-array-sum/problem}
 */
//---------- imports ----------
import java.util.*;


//---------- solution ----------
class Result
{

    /**
     * Complete the 'simpleArraySum' function below.
     * 
     * @param {List<Integer>} The function accepts INTEGER_ARRAY ar as parameter.
     * @return The function is expected to return an INTEGER.
     */

    public static int simpleArraySum(List<Integer> ar) 
    { return (ar.stream().reduce(0, Integer::sum)); }
}//end class

File: go/algorithms/easy/simpleArraySum.go
/*
Author: Rithie Natan Carvalhaes Prado
Date: 2026-04-25
Last modified: 2026-06-13
Description: Simple Array Sum - HackerRank
See: https://www.hackerrank.com/challenges/simple-array-sum/problem
*/
//--------- package ---------
package easy


//--------- functions ---------
/**
 * Complete the 'simpleArraySum' function below.
 *
 * @ar: The function accepts INTEGER_ARRAY ar as parameter.
 * @return: The function is expected to return an INTEGER.
 */
func simpleArraySum(ar []int32) int32 {
	var sum int32 = 0;

	for _, value := range ar {
		sum += value;
	}//end for

	return sum;
}//end simpleArraySum()